state-of-k8s-cost-optimization/kube-requests-checker.sh

```diff
@@ -0,0 +1,134 @@
+#!/bin/bash
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o pipefail
+
+# Setup variables
+
+CLI_NAME=${0}
+OPTIONS=$(getopt -o n:l:ih --long namespace:,level:,include-kube-system,help -- "$@")
+EXCLUSIONS="--field-selector metadata.namespace!=kube-system"
```

Comment:
nit: consider defining these variables as `local`

Reply:
can i do this outside of a bash function?